### PR TITLE
Use correct ref when finding directories in repos

### PR DESCRIPTION
**Pull request by mrnugget**
@eseliger found this potential bug. It's extracted from the other PR with the tests fixed.

### CODE REVIEW

File: internal/batches/service/service.go [hunk 1]
```diff
@@ -660,7 +660,7 @@ func (svc *Service) FindDirectoriesInRepos(ctx context.Context, fileName string,
 		a.WriteString("query DirectoriesContainingFile() {\n")
 
 		for _, repo := range batch {
-			query := fmt.Sprintf(`file:(^|/)%s$ repo:^%s$ type:path count:99999`, regexp.QuoteMeta(fileName), regexp.QuoteMeta(repo.Name))
+			query := fmt.Sprintf(`file:(^|/)%s$ repo:^%s$@%s type:path count:99999`, regexp.QuoteMeta(fileName), repo.BaseRef(), regexp.QuoteMeta(repo.Name))
```
Comment (eseliger) [inline internal/batches/service/service.go:663]: I'm a bit scared of spaces or whatever breaking this, but what can we do 🤷‍♂️

**Review by eseliger — APPROVED** [review]
Just to make sure: did you give this a try yet? I didn't actually try it out yet on that branch where it came from.

File: internal/batches/service/service.go (hunk 1 above)
Comment (mrnugget) [reply internal/batches/service/service.go:663]: Spaces in branch names?

Comment (eseliger) [reply internal/batches/service/service.go:663]: no idea what people can get into git.. :D

**Comment by mrnugget** [comment]
> Just to make sure: did you give this a try yet?

Yep, tried it locally.

**Pull request merged**